### PR TITLE
chore: Prettify and improve code markup

### PR DESCRIPTION
**Pull request by jackmu95**
## Description of Pull Request:

This PR prettifies the codebase again and cleans up some code examples for space optimization.

Also closes following issues:
#2484

## PR checklist:

**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix
- [ ] Feature
- [ ] Enhancement to an existing feature
- [ ] ARIA accessibility
- [x] Documentation update
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact:

**The PR fulfills these requirements:**

- [x] It's submitted to the `dev` branch, **not** the `master` branch
- [x] When resolving a specific issue, it's referenced in the PR's title (i.e.
      `fixes #xxxx[,#xxxx]`, where "xxxx" is the issue number)
- [x] The PR should address only one issue or feature. If adding multiple features or fixing a bug
      and adding a new feature, break them into separate PRs if at all possible.
- [x] PR titles should following the
      [**Conventional Commits**](https://www.conventionalcommits.org/) naming convention (i.e.
      "fix(alert): not alerting during SSR render", "docs(badge): Updated pill examples, fix typos",
      "chore: fix typo in docs", etc). **This is very important, as the `CHANGELOG` is generated
      from these messages.**

**If new features/enhancement/fixes are added or changed:**

- [ ] Includes documentation updates (including updating the component's `package.json` for slot and
      event changes)
- [ ] New/updated tests are included and passing (if required)
- [ ] Existing test suites are passing
- [ ] The changes have not impacted the functionality of other components or directives
- [ ] ARIA Accessibility has been taken into consideration (does it affect screen reader users or
      keyboard only users? clickable items should be in the tab index, etc)

**If adding a new feature, or changing the functionality of an existing feature, the PR's
description above includes:**

- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a
      suggestion issue first and wait for approval before working on it)

### CODE REVIEW

File: src/components/button-group/README.md [hunk 1]
```diff
@@ -6,20 +6,23 @@
...
-    <b-button variant="warning">Warning</b-button>
-  </b-button-group>
+  <div>
+    <b-button-group>
+      <b-button>Button 1</b-button>
+      <b-button>Button 2</b-button>
+      <b-button>Button 3</b-button>
+    </b-button-group>
+  </div>
+  <div class="mt-3">
```
Comment (jackmu95) [inline src/components/button-group/README.md]: @tmorehouse `<div class="mt-3">...</div>` vs. `<br><br>`?

Comment (tmorehouse) [reply src/components/button-group/README.md]: Either mt-3 or mb-3..

File: docs/components/codemirror.vue [hunk 2]
```diff
@@ -1,8 +1,5 @@
 <template>
-  <textarea
-    ref="textArea"
-    :value="value"
-  />
+  <textarea ref="textArea" :value="value" />
```
[the remark this answers is not in the document]
Comment (jackmu95) [reply docs/components/codemirror.vue]: We chose to allow up to `4` attributes per line to save some space in Vue templates - especially in the docs.

**Comment by jackmu95** [comment]
@tmorehouse Anything that is holding merging this back?

**Comment by tmorehouse** [comment]
@jackmu95 

Just a few intermediate PRs in dev.

Will need to update this branch with the changes.

**Comment by tmorehouse** [comment]
@jackmu95 anything else to tweak in this, or can we merge it now?

**Comment by jackmu95** [comment]
@tmorehouse LGTM!

**Pull request merged**